File: frontend/src/lib/schemas/market-conditions.ts
import { z } from 'zod';

export const marketConditionsSchema = z.object({
  /** Most recent Brent crude daily close (USD per barrel), or null */
  brent_price: z.number().nullable(),

  /** Previous Brent crude daily close — used for two-consecutive-closes Rule 3 check */
  brent_prev_price: z.number().nullable(),

  /** Latest CBOE VIX index level, or null */
  vix_value: z.number().nullable(),
});

export type MarketConditions = z.infer<typeof marketConditionsSchema>;

File: backend/src/atlas/schemas/market_conditions.py
"""Pydantic schema for the Market Conditions endpoint."""

from pydantic import BaseModel, Field


class MarketConditionsResponse(BaseModel):
    """Live market-regime inputs: Brent crude and VIX.

    Both values are None when the upstream data source is unavailable.
    ``brent_prev_price`` is the previous daily close, required by the frontend
    to evaluate the two-consecutive-closes test for Rule 3.
    """

    brent_price: float | None = Field(
        description="Most recent Brent crude daily close (USD per barrel).",
    )
    brent_prev_price: float | None = Field(
        description="Previous Brent crude daily close (USD per barrel). "
        "Used to check two consecutive closes below $95 for Rule 3.",
    )
    vix_value: float | None = Field(
        description="Latest CBOE VIX index level.",
    )
